python/lsst/obs/metadata/translator.py

```diff
@@ -0,0 +1,252 @@
+
+        Returns
+        -------
+        translator : `MetadataTranslator`
+            Translation class that knows how to extract metadata from
+            the supplied header.
+        """
+        for name, trans in cls.translators.items():
+            if trans.canTranslate(header):
+                log.debug(f"Using translation class {name}")
+                return trans
+        else:
+            raise ValueError("None of the registered translation classes understood this header")
+
+    def _used_these_cards(self, *args):
+        """Indicate that the supplied cards have been used for translation.
+
+        Parameters
+        ----------
+        args : sequence of `str`
+            Keywords used to process a translation.
+        """
+        self._used_cards.update(set(args))
+
+    def cards_used(self):
+        """Cards used during metadata extraction.
+
+        Returns
+        -------
+        used : `frozenset`
+            Cards used when extracting metadata.
+        """
+        return frozenset(self._used_cards)
+
+    @staticmethod
+    def validate_value(value, default, minimum=None, maximum=None):
+        """Validate the supplied value, returning a new value if out of range
+
+        Parameters
+        ----------
+        value : `float`
+            Value to be validated.
+        default : `float`
+            Default value to use if supplied value is invalid or out of range.
+            Assumed to be in the same units as the value expected in the
+            header.
+        minimum : `float`
+            Minimum possible valid value, optional.  If the calculated value
+            is below this value, the default value will be used.
+        maximum : `float`
+            Maximum possible valid value, optional.  If the calculated value
+            is above this value, the default value will be used.
+
+        Returns
+        -------
+        value : `float`
+            Either the supplied value, or a default value.
+        """
+        print(f"Validating {value} {default}")
+        if math.isnan(value):
+            value = default
+        else:
+            if minimum is not None and value < minimum:
+                value = default
+            elif maximum is not None and value > maximum:
+                value = default
+        return value
+
+    def quantity_from_card(self, keyword, unit, scaling=1., default=None, minimum=None, maximum=None):
+        """Calculate a Astropy Quantity from a header card and a unit.
+
+        Parameters
+        ----------
+        keyword : `str`
+            Keyword to use from header.
+        unit : `astropy.units.UnitBase`
+            Unit of the item in the header.
+        scaling : `float`
+            Amount to scale the value from the header before assigning the
+            units.  This is necessary if Astropy does not support the unit
+            directly (such as for mbar pressure units). The scaling is applied
```

Comment:
Is it worth passing those missing units back up to astropy, so they can support them in the future?

Reply:
Yes. See astropy/astropy#7863

I have switched to `hPa` though so I think I should remove the scaling hack here.